cdap-proto/src/main/java/co/cask/cdap/proto/ProtoTriggerCodec.java

```diff
@@ -83,7 +83,13 @@ public Trigger deserialize(JsonElement json, Type typeOfT, JsonDeserializationCo
     }
     JsonObject object = (JsonObject) json;
     JsonElement typeJson = object.get("type");
-    ProtoTrigger.Type triggerType = context.deserialize(typeJson, ProtoTrigger.Type.class);
+    ProtoTrigger.Type triggerType;
+    try {
+      String typeString = typeJson.getAsString();
+      triggerType = ProtoTrigger.Type.valueOf(typeString.toUpperCase());
+    } catch (Exception e) {
```

Comment:
It might, but there may be a NullPointerException, too, and getAsString() can also throw something else. These would bubble up without explanation.